crates/bevy_core_pipeline/src/bloom/mod.rs

```diff
@@ -204,33 +207,80 @@ impl Node for BloomNode {
 
         let pipelines = world.resource::<BloomPipelines>();
         let pipeline_cache = world.resource::<PipelineCache>();
+        let bloom_uniforms = world.resource::<BloomUniforms>();
         let view_entity = graph.get_input_entity(Self::IN_VIEW)?;
-        let (camera, view_target, textures, bind_groups, uniform_index) =
-            match self.view_query.get_manual(world, view_entity) {
-                Ok(result) => result,
-                _ => return Ok(()),
-            };
         let (
-            downsampling_prefilter_pipeline,
-            downsampling_pipeline,
-            upsampling_pipeline,
-            upsampling_final_pipeline,
-        ) = match (
-            pipeline_cache.get_render_pipeline(pipelines.downsampling_prefilter_pipeline),
+            Ok((camera, view_target, bloom_texture, bind_groups, uniform_index)),
+            Some(downsampling_first_pipeline),
+            Some(downsampling_pipeline),
+            Some(upsampling_pipeline),
+            Some(upsampling_final_pipeline),
+            Some(bloom_uniforms),
+        ) = (
+            self.view_query.get_manual(world, view_entity),
+            pipeline_cache.get_render_pipeline(pipelines.downsampling_first_pipeline),
             pipeline_cache.get_render_pipeline(pipelines.downsampling_pipeline),
             pipeline_cache.get_render_pipeline(pipelines.upsampling_pipeline),
             pipeline_cache.get_render_pipeline(pipelines.upsampling_final_pipeline),
-        ) {
-            (Some(p1), Some(p2), Some(p3), Some(p4)) => (p1, p2, p3, p4),
-            _ => return Ok(()),
+            bloom_uniforms.buffer.binding(),
+        ) else {
+            return Ok(());
         };
+        let view_target = view_target.post_process_write();
+
+        let downsampling_first_bind_group =
+            render_context
+                .render_device
+                .create_bind_group(&BindGroupDescriptor {
```

Comment:
is there a reason why bind_groups are created here? I thought bind_groups were supposed to be created in the Prepare phase.

Reply:
It has to be created here, because it's binding to the view_target.post_process_write() result, which has to be created in Node::run(). The rest of the bind groups that don't do this, I create in the prepare phase like normal.